braindecode/augmentation/base.py

```diff
@@ -0,0 +1,211 @@
+        to another argument of object with more meaning. In case both are
+        passed, magnitude will override the latter. Defaults to None.
+    mag_range : tuple of two floats | None, optional
+        Valid range of the argument mapped by `magnitude` (e.g. standard
+        deviation, number of sample, etc.):
+        ```
+        argument = magnitude * mag_range[1] + (1 - magnitude) * mag_range[0].
+        ```
+        If `magnitude` is None it is ignored. Defaults to None.
+    random_state: int, optional
+        Seed to be used to instatiate numpy random number generator instance.
+        Used to decide whether or not to transform given the probability
+        argument. Defaults to None.
+    *args:
+        Arguments to be passed to operation.
+    **kwargs:
+        Keyword arguments to be passed to operation.
+    """
+
+    def __init__(self, operation, probability=1.0, magnitude=None,
+                 mag_range=None, random_state=None, *args, **kwargs):
+        super().__init__()
+        assert callable(operation), "operation should be a `callable`."
+        self.operation = operation
+        assert isinstance(probability, Real), (
+            f"probability should be a `real`. Got {type(probability)}.")
+        assert probability <= 1. and probability >= 0., \
+            "probability should be between 0 and 1."
+        self._probability = probability
+        self.rng = check_random_state(random_state)
+        assert (
+            (
+                isinstance(magnitude, Real) and
+                0 <= magnitude <= 1
+            ) or magnitude is None
+        ), "magnitude can be either a float between 0 and 1 or None"
+        self._magnitude = magnitude
+        assert (
+            (
+                isinstance(mag_range, Iterable) and
+                len(mag_range) == 2 and
+                all([isinstance(v, Real) for v in mag_range])
+            ) or mag_range is None
+        ), "mag_range should be None or a tuple of two floats."
+        self.mag_range = mag_range
+        self.args = args
+        self.kwargs = kwargs
+
+    def forward(self, X: Tensor, y: Tensor) -> Output:
+        mask = self._get_mask(X.shape[0])
+        tr_X, tr_y = self.operation(
+            X.clone(), y.clone(), *self.args,
+            random_state=self.rng, magnitude=self.magnitude, **self.kwargs)
+
+        mask.squeeze_()
+        out_X, out_y = X.clone(), y.clone()
+        num_valid = mask.sum().long()
+        if num_valid > 0:
+            out_X[mask == 1, ...] = tr_X[mask == 1, ...]
+            if type(tr_y) is tuple:
+                out_y = tuple(tmp_y[mask == 1] for tmp_y in tr_y)
+            else:
+                out_y[mask == 1] = tr_y[mask == 1]
+        return out_X, out_y
+
+    def _get_mask(self, batch_size=None) -> torch.Tensor:
+        """Samples whether to apply operation or not over the whole batch
+        """
+        size = (batch_size, 1, 1)
```

Comment:
why do you need the two empty dimensions at end?

Reply:
It comes from a more complex version of Transform but it's indeed useless here. Removing it.